code/modules/tgui_input/say_modal/speech.dm

```diff
@@ -44,6 +44,14 @@
 		if(OOC_CHANNEL)
 			client.ooc(entry)
 			return TRUE
+		if(ADMIN_CHANNEL)
+			if(!check_rights_for(client, R_ADMIN))
+				// there's probably something downstream that handles this, but
+				// if it breaks at some point it's nice to have insurance
+				return;
```

Comment:
dynamic does check for permission rights, but I see no harm if having this check